mesh/src/main/java/no/nordicsemi/android/mesh/transport/LowerTransportLayer.java

```diff
@@ -484,13 +484,36 @@ final AccessMessage parseSegmentedAccessLowerTransportPDU(final int ttl,
             MeshLogger.verbose(TAG, "Starting incomplete timer for src: " + MeshAddress.formatAddress(blockAckDst, false));
             initIncompleteTimer();
 
-            //Start acknowledgement calculation and timer only for messages directed to a unicast address.
+            // Start acknowledgement calculation and timer only for messages directed to a unicast address.
             if (MeshAddress.isValidUnicastAddress(dst)) {
                 // Calculate the initial block acknowledgement value
                 mSegmentedAccessBlockAck = BlockAcknowledgementMessage.calculateBlockAcknowledgement(null, segO);
                 //Start the block acknowledgement timer irrespective of which segment was received first
                 initSegmentedAccessAcknowledgementTimer(seqZero, ttl, blockAckSrc, blockAckDst, segN);
             }
+
+            // We need to ensure there could be an unsegmented message that could be sent as a
+            // segmented message i.e. segO = 0 and segN = 0
+            if (segO == segN) {
```

Comment:
I think this will introduce another bug. It may happen that for multisegment message, you might receive the very last segment first due to multi-path propogation in the network, and then "SegO == segN" will become true for the first received segment itself. This will stop the recepion.

Instead, I would say you could do: !SegO and !SegN, to specifically process the case of "SegO = SegN = 0".

Reply:
Good point!